LibGit2Sharp/Branch.cs

```diff
@@ -226,7 +236,17 @@ private Branch ResolveTrackedBranch()
                 return null;
             }
 
-            string trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            string trackedReferenceName = null;
+            try
+            {
+                trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            }
+            catch (InvalidSpecificationException)
+            {
+                // we have no way to pro-actively check this
```

Comment:
Hmm could we check if `CanonicalName` is a valid Uri by converting it into a UriString and checking `IsValidAddress`? And then only squelching if that's the case? Or is that crazy?